[DPE-5565] Release notes for 195/149 release

deusebio opened the pull request:
Release notes for the upcoming release. 

I would still like to tune a bit the community message and some other bits around. @grobbie maybe you can craft the main message for the community.

izmalk reviewed (commented):
These are some suggestions on how to improve the existing release notes. Feel free to apply them how you see fit. We can discuss the new structure on the sprint or after.
If no structural changes are wanted, then I'd recommend just adding empty lines after headings, renaming "General updates" sections, and moving some of its contents to the Notes section at the end.

deusebio commented on docs/reference/r-releases/r-rev185_149.md:
```diff
@@ -0,0 +1,96 @@
+## Kafka
+
+### Features
+* [DPE-2285] Refer to Charmhub space from GitHub (#200)
```
I would leave it as is, as it is the name of the PR and it would be more confusing otherwise.

izmalk replied:
We talked to @a-velasco and as far as I understood, she pays a lot of attention to rephrasing entries to release notes to make them more meaningful. 
I don't see a big problem with these entries being renamed as long as the link to the PR is there. But if you think it's confusing, can't we just rename the PR accordingly?

a-velasco replied:
@deusebio For some context, since we don't have separate release notes for high-level users of the charm and for developers, I try to make them useful for both. This means rephrasing some of the PR titles so that someone can understand what was done just by reading that sentence, rather than having to do a deep dive into GitHub/Jira to figure out if it's relevant to them or not.

I think this extra effort could be mitigated in two ways:
1. Encouraging more meaningful PR titles (admittedly not so easy to do, but we would provide guidance and examples)
2. Separating user release notes from developer release notes. Dev release notes can be easily auto-generated from the commits; the question is just where to put them (e.g. CHANGELOG.md, Discourse..?). User release notes could remain on Discourse and focus on a just few major highlights for that release with a bit of explanation, links to docs, etc.

deusebio replied:
As I was saying to @izmalk, right now there is a nice correspondence between Release notes, PR Titles and commits. We can rename PRs but we can't rename commits. Also sometimes (when using rebase and merge) there may exists commits without the PR number. That's why I would find confusing to break this correspondence. 

Anyhow, my point of view is that rather than changing the release notes, we should nudge ourself to have meaningful PR titles. I believe we are already aiming at that (I don't think that there are engineers purposely putting misleading titles or descriptions), but what is meaningful is somewhat subjective. Providing guidance could definitely help to improve this. So, overall, I'd rather fix that over changing the names.

I'm not against having high-level user release notes, but this is probably for product managers to craft and have a call on.

deusebio commented on docs/reference/r-releases/r-rev185_149.md:
```diff
@@ -0,0 +1,96 @@
+
+## Inside the charms
+
+* Charmed ZooKeeper charm ships the ZooKeeper [3.8.4-ubuntu0](https://launchpad.net/zookeeper-releases/3.x/3.8.4-ubuntu0), built and supported by Canonical
+* Charmed Kafka charm ships the Kafka [3.6.1-ubuntu0](https://launchpad.net/kafka-releases/3.x/3.6.1-ubuntu0), built and supported by Canonical
+* Charmed ZooKeeper charm is based on [charmed-zookeeper snap](https://snapcraft.io/charmed-zookeeper) on the `3/stable` (Ubuntu LTS “22.04” - core22-based)
+* Charmed Kafka charm is based on [charmed-kafka snap](https://snapcraft.io/charmed-kafka) on the `3/stable` channel (Ubuntu LTS “22.04” - core22-based)
+* Principal charms supports the latest LTS series “22.04” only.
```
Just keeping consistent with the previous release notes

izmalk replied:
I'd say this is very well worth the improvement, but OK, it's up to you.

deusebio commented:
@izmalk I have addressed the comments whenever it was sensible to do, while keeping consistency with the previous release notes for a similar look and feel. 

I would leave it up to you and Andreia if we want to change the format. But when doing that, we should probably reformat also the other pages, always keeping a similar style across the documentation

deusebio commented on docs/reference/r-releases/r-rev185_149.md:
```diff
@@ -0,0 +1,109 @@
+Dear community,
+
+We are glad to report that we have just released a new updated version for the Charmed Kafka bundles on the `3/stable` channel, 
+upgrading Charm ZooKeeper revision from 136 to 149, and Charmed Kafka from xxx to 188.  
```
yes, once we will finalize the release (we are still missing a small fix), I will update this

deusebio commented on docs/reference/r-releases/r-rev185_149.md:
```diff
@@ -0,0 +1,96 @@
+* Charmed Kafka charm is based on [charmed-kafka snap](https://snapcraft.io/charmed-kafka) on the `3/stable` channel (Ubuntu LTS “22.04” - core22-based)
+* Principal charms supports the latest LTS series “22.04” only.
+
+More information about the artifacts are provided by the following table:
```
I'd keep it as in the previous release notes for consistency.

deusebio commented on docs/reference/r-releases/r-rev185_149.md:
```diff
@@ -0,0 +1,109 @@
+
+## Kafka
+
+Here below we provide a summary of the main features and bug fixes related to the Kafka operator.
```
Uhm, IMHO this sounds a bit of a repetition from the list below, for which I'm not sure I see the value apart from not having section and subsection next to each other (which I didn't know it wasn't allowed)

izmalk replied:
Well, a summary is a short recollection, like a TL;DR. But given the original list of features and bugfixes is well structured and not that long, it may be not that important to provide a summary. At least let's make the introduction shorter, since it doesn't bring much in its current form. I'm suggesting the following:
`New features and bug fixes in the Charmed Apache Kafka operator:`
Not a blocker.

deusebio replied:
sounds good!

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+
+Please reach out should you have any question, comment, feedback or information. You can find us here in [Matrix](https://matrix.to/#/#charmhub-data-platform:ubuntu.com) or also on [Discourse](https://discourse.charmhub.io/).
+
+## Kafka
```
```suggestion
## Apache Kafka
```

deusebio replied:
I used `Charmed Apache Kafka` since we are talking about the operator here

izmalk replied:
**note**: That potentially creates a recursion - The release notes are for the Charmed Apache Kafka solution, so I'd avoid using the same name for one of its headings/components. 
Alternatively, we can update the heading to be about something more specific. For example, not the solution but the operator or bundle: `Charmed Apache Kafka operator`.

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+* [discourse-gatekeeper] Migrate charm docs (#210, #203, #198, #194, #192)
+* [DPE-3932] Update information in metadata.yaml
+
+## ZooKeeper
```
```suggestion
## Apache ZooKeeper
```

deusebio replied:
I used `Charmed Apache ZooKeeper`

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+
+## Kafka
+
+New features and bug fixes in the Charmed Apache Kafka operator:
```
Please, double-check that we want to talk specifically about the operator only here.

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+
+## Inside the charms
+
+Contents of the Kafka and ZooKeeper charms include:
```
```suggestion
Contents of the Apache Kafka and Apache ZooKeeper charms include:
```

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+Contents of the Kafka and ZooKeeper charms include:
+
+* Charmed ZooKeeper charm is based on the [charmed-zookeeper snap](https://snapcraft.io/charmed-zookeeper) of the `3/stable` channel (Ubuntu LTS “22.04” - core22-based) that ships the ZooKeeper [3.8.4-ubuntu0](https://launchpad.net/zookeeper-releases/3.x/3.8.4-ubuntu0), built and supported by Canonical
+* Charmed Kafka charm is based on the [charmed-kafka snap](https://snapcraft.io/charmed-kafka) of the `3/stable` channel (Ubuntu LTS “22.04” - core22-based) that ships the Kafka [3.6.1-ubuntu0](https://launchpad.net/kafka-releases/3.x/3.6.1-ubuntu0), built and supported by Canonical
```
It's either `Charmed Apache Kafka` or `Apache Kafka charm`.
```suggestion
* Charmed Apache Kafka is based on the [charmed-kafka snap](https://snapcraft.io/charmed-kafka) of the `3/stable` channel (Ubuntu LTS “22.04” - core22-based) that ships the Apache Kafka [3.6.1-ubuntu0](https://launchpad.net/kafka-releases/3.x/3.6.1-ubuntu0), built and supported by Canonical
```

deusebio replied:
Let's go with `Charmed Apache Kafka`

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+| Artifact               | Track/Series | Version/Revision | Code                                                                                                                |
+|------------------------|--------------|------------------|---------------------------------------------------------------------------------------------------------------------|
+| ZooKeeper distribution | 3.x          | 3.8.4-ubuntu0    | [78499c](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.4&id=78499c9f4d4610f9fb963afdad1ffd1aab2a96b8) |
+| Kafka distribution     | 3.x          | 3.6.1-ubuntu0    | [db44db](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.1&id=db44db1ebf870854dddfc3be0187a976b997d4dc)     |
```
```suggestion
| Apache Kafka distribution     | 3.x          | 3.6.1-ubuntu0    | [db44db](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.1&id=db44db1ebf870854dddfc3be0187a976b997d4dc)     |
```

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+| Kafka distribution     | 3.x          | 3.6.1-ubuntu0    | [db44db](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.1&id=db44db1ebf870854dddfc3be0187a976b997d4dc)     |
+| Charmed ZooKeeper snap | 3/stable     | 34               | [13f3c6](https://github.com/canonical/charmed-zookeeper-snap/tree/13f3c620658fdc55b7d6745b81c7b5a00e042e10)         |        
+| ZooKeeper operator     | 3/stable     | 149              | [40576c](https://github.com/canonical/zookeeper-operator/commit/40576c1c87badd1e2352afc013ed0754808ef44c)           | 
+| Charmed Kafka snap     | 3/stable     | 37               | [c266f9](https://github.com/canonical/charmed-kafka-snap/tree/c266f9cd283408d2106d4682b67661205a12ea7f)             |  
```
```suggestion
| Charmed Apache Kafka snap     | 3/stable     | 37               | [c266f9](https://github.com/canonical/charmed-kafka-snap/tree/c266f9cd283408d2106d4682b67661205a12ea7f)             |  
```

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+| Charmed ZooKeeper snap | 3/stable     | 34               | [13f3c6](https://github.com/canonical/charmed-zookeeper-snap/tree/13f3c620658fdc55b7d6745b81c7b5a00e042e10)         |        
+| ZooKeeper operator     | 3/stable     | 149              | [40576c](https://github.com/canonical/zookeeper-operator/commit/40576c1c87badd1e2352afc013ed0754808ef44c)           | 
+| Charmed Kafka snap     | 3/stable     | 37               | [c266f9](https://github.com/canonical/charmed-kafka-snap/tree/c266f9cd283408d2106d4682b67661205a12ea7f)             |  
+| Kafka operator         | 3/stable     | 195              | [7948df](https://github.com/canonical/kafka-operator/pull/241/commits/7948dfbbfaaa53fccc88beaa90f80de1e70beaa9)                 |   
```
```suggestion
| Charmed Apache Kafka operator         | 3/stable     | 195              | [7948df](https://github.com/canonical/kafka-operator/pull/241/commits/7948dfbbfaaa53fccc88beaa90f80de1e70beaa9)                 |   
```

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+
+Contents of the Kafka and ZooKeeper charms include:
+
+* Charmed ZooKeeper charm is based on the [charmed-zookeeper snap](https://snapcraft.io/charmed-zookeeper) of the `3/stable` channel (Ubuntu LTS “22.04” - core22-based) that ships the ZooKeeper [3.8.4-ubuntu0](https://launchpad.net/zookeeper-releases/3.x/3.8.4-ubuntu0), built and supported by Canonical
```
Same as below:
```suggestion
* Charmed Apache ZooKeeper is based on the [charmed-zookeeper snap](https://snapcraft.io/charmed-zookeeper) of the `3/stable` channel (Ubuntu LTS “22.04” - core22-based) that ships the Apache ZooKeeper [3.8.4-ubuntu0](https://launchpad.net/zookeeper-releases/3.x/3.8.4-ubuntu0), built and supported by Canonical
```

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+
+| Artifact               | Track/Series | Version/Revision | Code                                                                                                                |
+|------------------------|--------------|------------------|---------------------------------------------------------------------------------------------------------------------|
+| ZooKeeper distribution | 3.x          | 3.8.4-ubuntu0    | [78499c](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.4&id=78499c9f4d4610f9fb963afdad1ffd1aab2a96b8) |
```
```suggestion
| Apache ZooKeeper distribution | 3.x          | 3.8.4-ubuntu0    | [78499c](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.4&id=78499c9f4d4610f9fb963afdad1ffd1aab2a96b8) |
```

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+|------------------------|--------------|------------------|---------------------------------------------------------------------------------------------------------------------|
+| ZooKeeper distribution | 3.x          | 3.8.4-ubuntu0    | [78499c](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.4&id=78499c9f4d4610f9fb963afdad1ffd1aab2a96b8) |
+| Kafka distribution     | 3.x          | 3.6.1-ubuntu0    | [db44db](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.1&id=db44db1ebf870854dddfc3be0187a976b997d4dc)     |
+| Charmed ZooKeeper snap | 3/stable     | 34               | [13f3c6](https://github.com/canonical/charmed-zookeeper-snap/tree/13f3c620658fdc55b7d6745b81c7b5a00e042e10)         |        
```
```suggestion
| Charmed Apache ZooKeeper snap | 3/stable     | 34               | [13f3c6](https://github.com/canonical/charmed-zookeeper-snap/tree/13f3c620658fdc55b7d6745b81c7b5a00e042e10)         |        
```

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,108 @@
+| ZooKeeper distribution | 3.x          | 3.8.4-ubuntu0    | [78499c](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.4&id=78499c9f4d4610f9fb963afdad1ffd1aab2a96b8) |
+| Kafka distribution     | 3.x          | 3.6.1-ubuntu0    | [db44db](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.1&id=db44db1ebf870854dddfc3be0187a976b997d4dc)     |
+| Charmed ZooKeeper snap | 3/stable     | 34               | [13f3c6](https://github.com/canonical/charmed-zookeeper-snap/tree/13f3c620658fdc55b7d6745b81c7b5a00e042e10)         |        
+| ZooKeeper operator     | 3/stable     | 149              | [40576c](https://github.com/canonical/zookeeper-operator/commit/40576c1c87badd1e2352afc013ed0754808ef44c)           | 
```
```suggestion
| Charmed Apache ZooKeeper operator     | 3/stable     | 149              | [40576c](https://github.com/canonical/zookeeper-operator/commit/40576c1c87badd1e2352afc013ed0754808ef44c)           | 
```

izmalk reviewed (changes requested):
Looks good for the older comments but we need to take into account the newest changes in terminology.

deusebio commented:
@izmalk thanks for spotting this. I have applied the changes, please check [85c249f](https://github.com/canonical/kafka-operator/pull/263/commits/85c249f2516776a26b98ee81247f69c340c06c9c)

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,109 @@
+
+## Charmed Apache ZooKeeper
+
+New features and bug fixes in the Charmed Apache ZooKeeper:
```
```suggestion
New features and bug fixes in the Charmed Apache ZooKeeper bundle:
```
See above

izmalk reviewed (commented):
I like the structure that we have now but I have a non-blocking suggestion for the naming of components.
Also, one Apache Kafka renaming is missing.

izmalk commented on docs/reference/r-releases/r-rev195_149.md:
```diff
@@ -0,0 +1,109 @@
+
+Contents of the Charmed Apache Kafka and Charmed Apache ZooKeeper include:
+
+* Charmed Apache ZooKeeper is based on the [charmed-zookeeper snap](https://snapcraft.io/charmed-zookeeper) of the `3/stable` channel (Ubuntu LTS “22.04” - core22-based) that ships the ZooKeeper [3.8.4-ubuntu0](https://launchpad.net/zookeeper-releases/3.x/3.8.4-ubuntu0), built and supported by Canonical
```
```suggestion
* Charmed Apache ZooKeeper is based on the [charmed-zookeeper snap](https://snapcraft.io/charmed-zookeeper) of the `3/stable` channel (Ubuntu LTS “22.04” - core22-based) that ships the Apache ZooKeeper [3.8.4-ubuntu0](https://launchpad.net/zookeeper-releases/3.x/3.8.4-ubuntu0), built and supported by Canonical
```

deusebio commented:
Content has been copied to Discourse [here](https://discourse.charmhub.io/t/charmed-apache-kafka-documentation-reference-release-195-149/16140)
